tutorials/parameter_sweep_demo.ipynb

```diff
@@ -0,0 +1,783 @@
+   "metadata": {
+    "slideshow": {
+     "slide_type": "subslide"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "pprint(results_array)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7d14aba1",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Plotting Results\n",
+    "\n",
+    "The resulting H5 files can then be prcessed to generate plots, e.g., 2D scatter plots, map plots, etc.\n",
+    "\n",
+    "<table><tr>\n",
+    "    <td> <img src=\"assets_parameter_sweep_demo/scatter_LCOW_vs_Acomp.jpg\" alt=\"LCOW vs water permeability\" width=\"100%\"/> </td>\n",
+    "    <td> <img src=\"assets_parameter_sweep_demo/maps_LCOW_recovery_NaCl_loading.jpg\" alt=\"LCOW vs water recovery and NaCL concentration\" width=\"100%\"/> </td>\n",
+    "</tr></table>\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d95eb8b4",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "# Advanced Usage and Features"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4ea976e5",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Types of Parameter Sweeps\n",
+    "\n",
+    "* In its current form, a user can run 3 types of parameter sweeps\n",
+    "    - Simple parameter sweep\n",
+    "    - **Recursive parameter sweep**\n",
+    "    - Differential parameter sweep\n",
+    "* Samples for the parameter sweep can be generated from a probability distribution or an *n*-dimensional euclidean space where *n* is the number of parameters in the sweep. Latin hypercube sampling is also enabled."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f8f87e0c",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "subslide"
+    }
+   },
+   "source": [
+    "### Recursive Parameter Sweep\n",
+    "\n",
+    "\n",
+    "<div>\n",
+    "    <img src=\"assets_parameter_sweep_demo/Recursive_Parameter_Sweep_flowchart.png\" alt=\"RecursiveParameterSweep\" width=\"100%\" align=\"center\"/>\n",
+    "</div>\n",
+    "\n",
+    "*Run the simple parameter sweep in recursion if some runs fail to ensure that a user-specified number of sample results are generated. This involves resampling the input parameter space to compensate for the failed runs.*"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "01aa99e4",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Types of Parameter Sweeps\n",
+    "\n",
+    "* In its current form, a user can run 3 types of parameter sweeps\n",
+    "    - Simple parameter sweep\n",
+    "    - Recursive parameter sweep\n",
+    "    - **Differential parameter sweep**\n",
+    "* Samples for the parameter sweep can be generated from a probability distribution or an *n*-dimensional euclidean space where *n* is the number of parameters in the sweep. Latin hypercube sampling is also enabled."
+   ]
```

Comment:
I think we should add the simplest (most concise) code block possible to demonstrate exactly how to call/run the differential parameter sweep.

Reply:
Same here. This was a good exercise. I found more things to fix doing this.